src/python/pants/build_graph/target.py

```diff
@@ -113,16 +113,16 @@ def check_unknown(self, target, kwargs, payload):
       """
       ignore_params = set((self.get_options().ignored or {}).get(target.type_alias, ()))
       unknown_args = {arg: value for arg, value in kwargs.items() if arg not in ignore_params}
-      if 'source' in unknown_args:
-        if 'sources' in payload.as_dict():
-          deprecated_conditional(
-            lambda: True,
-            '1.10.0.dev0',
-            ('The source argument in targets is deprecated - it gets automatically promoted to '
-            'sources. Target {} should just use a sources argument. No BUILD files need changing. '
-            'The source argument will stop being populated -').format(target.type_alias),
-          )
-          unknown_args.pop('source')
+      if 'source' in kwargs and 'source' not in unknown_args and 'source' not in ignore_params:
```

Comment:
I'm having real difficulty parsing this line (could just be the time) -- it seems like `'source' in kwargs and 'source' not in unknown_args and 'source' not in ignore_params` is always going to be false, given the definition of `unknown_args` just above? Not trying to nitpick -- if there's something I'm missing let me know, I just feel like a very short comment describing the condition being checked here could be helpful in the future (since the point of this diff is to refactor that condition).

Reply:
Ugh, you're right, this isn't ever actually going to be hit. I'm just going to remove the deprecation warning, and cross my fingers...